### PR TITLE
SRCH-3125, SRCH-3126 Remove Google code, tests, db columns, fixtures, etc.

### PR DESCRIPTION
**Pull request by lorawoodford**
## Summary
- Continuation of the work started in #954;
- That initial branch was stopped due to code coverage issues, so new tests were added (`spec/models/api_docs_search_spec.rb`), updated (`spec/engines/api_cache_spec.rb` and `spec/models/api_azure_composite_web_search_spec.rb`), and unskipped (`spec/models/api_azure_search_spec.rb`) to get us back to 100% code coverage
- In addition to the 36 files changed in the initial PR, I've also removed unneeded fixtures and vcr cassettes (in two separate commits);
- And, finally, I've removed the `cx` param and supporting tests as that is only utilized by Google.

The 8 commits are intended to make review somewhat simpler as they logically group together stages of the removal.  Inline comments are also in place to help contextualize some changes.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

File: spec/models/api_azure_composite_web_search_spec.rb [hunk 1]
```diff
@@ -17,26 +17,38 @@
...
+        search
+        expect(BingV5WebEngine).to have_received(:new)
       end
 
       it 'uses module tag BV5W' do
-        expect(subject.default_module_tag).to eq('BV5W')
+        expect(search.default_module_tag).to eq('BV5W')
+      end
+
+      it 'returns a web hash' do
```
Comment (lorawoodford) [inline spec/models/api_azure_composite_web_search_spec.rb:50]: Addition of this test (for code coverage) is the only real change here... the rest are CodeClimate inspired style updates.

File: spec/models/api_azure_search_spec.rb [hunk 2]
```diff
@@ -365,35 +363,47 @@
       end
     end
 
-    skip '#as_json' do
+    describe '#as_json' do
```
Comment (lorawoodford) [inline spec/models/api_azure_search_spec.rb:366]: Unskipped this block for code coverage, and moved skips/explanations into specific describe blocks.

File: spec/models/api_azure_search_spec.rb [hunk 3]
```diff
@@ -365,35 +363,47 @@
...
+        # adequate code coverage during the removal of Google.
+        results = Rails.root.join('spec/fixtures/json/bing_v7/web_search/gov.json').read
+        stub_request(:get, /#{BingV5WebEngine::API_HOST}.*gov/o).
+          to_return(status: 200, body: results)
+        search.run
+      end
 
       it 'returns results' do
-        expect(search.as_json[:web][:results].count).to eq(20)
+        result = Hashie::Mash.new(search.as_json[:web][:results].first)
```
Comment (lorawoodford) [inline spec/models/api_azure_search_spec.rb:386]: Added for code coverage.  This and ln 366 are the only real changes here, the rest is CodeClimate inspired style changes.

**Comment by krbhavith** [comment]
@lorawoodford I think we need to remove this line related to GSS https://github.com/lorawoodford/search-gov/blob/SRCH-3125/config/routes.rb#L24

**Comment by lorawoodford** [comment]
> @lorawoodford I think we need to remove this line related to GSS https://github.com/lorawoodford/search-gov/blob/SRCH-3125/config/routes.rb#L24

Awesome, great catch.  Change made.  Thanks!

**Review by krbhavith — APPROVED** [review]
LGTM

**Pull request merged**